src/scripts/main.js

```diff
@@ -1 +1,66 @@
 'use strict';
+
+document.addEventListener('contextmenu', (e) => {
+  e.preventDefault();
+});
+
+const promise1 = new Promise((resolve, reject) => {
+  document.addEventListener('click', (e) => {
+    if (e.button === 0) {
+      resolve('First promise was resolved');
+    }
+  });
+
+  setTimeout(() => reject(new Error('First promise was rejected')), 3000);
```

Comment:
Consider increasing the timeout duration for the first promise rejection to ensure users have enough time to trigger the click event before it gets rejected.